dateutil/rrule.py

```diff
@@ -798,6 +844,27 @@ def _iter(self):
         total = 0
         count = self._count
         while True:
+            leap = False
+
+            # SKIP rule
+            if day > 28 and self._skip != 'OMIT':
+                daysinmonth = calendar.monthrange(year, month)[1]
+                if day > daysinmonth:
+                    if self._skip == 'BACKWARD':
+                        day = daysinmonth
+                        leap = True
+                    elif self._skip == 'FORWARD':
+                        day = 1
+                        month += 1
+                        if month == 13:
+                            month = 1
```

Comment:
I believe it's not actually possible to hit this case in the Gregorian calendar when only allowing skip condition 3, since December always has 31 days.

I'd have to check the RFC rules to see if, with the new extension, you can do something like `bymonthday=32` and have it fall forward. I suspect that this is unnecessary, since that should be equivalent to a `bymonthday=1` rule.

If this case is forbidden by the RFC (or if this set of conditions is only ever reached for end-of-month skip conditions), just drop this whole `if` condition and replace it with this:

```python
# In the Gregorian calendar, December has 31 days, so we should never find
# a skip condition that would roll over to the next year
assert month <= 12, "Skip forward logic resulted in anomalous month"
```